repository.go

```diff
@@ -977,29 +991,6 @@ func (r *Repository) ResolveRevision(rev plumbing.Revision) (*plumbing.Hash, err
 				return &plumbing.ZeroHash, fmt.Errorf(`No commit message match regexp : "%s"`, re.String())
 			}
 
-			commit = c
-		case revision.AtDate:
-			history := object.NewCommitPreorderIter(commit, nil, nil)
-
-			date := item.(revision.AtDate).Date
-
-			var c *object.Commit
-			err := history.ForEach(func(hc *object.Commit) error {
-				if date.Equal(hc.Committer.When.UTC()) || hc.Committer.When.UTC().Before(date) {
-					c = hc
-					return storer.ErrStop
-				}
-
-				return nil
-			})
-			if err != nil {
-				return &plumbing.ZeroHash, err
-			}
-
-			if c == nil {
-				return &plumbing.ZeroHash, fmt.Errorf(`No commit exists prior to date "%s"`, date.String())
-			}
-
```

Comment:
I removed this part because it's a bad implementation,I misread the spec, it must follows this : `This suffix may only be used immediately following a ref name and the ref must have an existing log ($GIT_DIR/logs/<ref>)`